mlserver/env.py

```diff
@@ -0,0 +1,74 @@
+import asyncio
+import os
+import sys
+import tarfile
+import glob
+
+from typing import List
+from functools import cached_property
+
+from .logging import logger
+
+
+def _extract_env(tarball_path: str, env_path: str) -> None:
+    logger.info(f"Extracting environment tarball from {tarball_path}...")
+    with tarfile.open(tarball_path, "r") as tarball:
+        tarball.extractall(path=env_path)
+
+
+class Environment:
+    """
+    Custom Python environment.
+    The class can be used as a context manager to enable / disable the custom
+    environment.
+    """
+
+    def __init__(self, env_path: str):
+        self._env_path = env_path
+
+    @classmethod
+    async def from_tarball(cls, tarball_path: str, env_path: str) -> "Environment":
+        loop = asyncio.get_running_loop()
+        await loop.run_in_executor(None, _extract_env, tarball_path, env_path)
+
+        return cls(env_path)
```

Comment:
not sure if this is part of this class but we probably want to make sure that this is a valid env?

Reply:
It's hard to check whether something is a valid env. However, the class should already handle gracefully when some of the environment base folders are missing.